src/mavlink_vehicles.cc

```diff
@@ -1002,6 +1002,9 @@ void mav_vehicle::takeoff()
 {
     switch (this->autopilot) {
     case autopilot_type::PX4:
+        if (this->get_arm_status() != arm_status::ARMED) {
+            this->arm_throttle();
```

Comment:
I think we should check the arm status again to be sure it's armed. Arming does a ton of safety checks and frequently it won't arm on a real drone. If we fail after explicit trying to arm, then I think we should issue an error. What do you think?